### PR TITLE
Add cookie banner 

### LINKED ISSUE #2692: Include cookie banner
> We have received guidance from our legal team that we need to include a cookie banner on all Mozilla websites that use cookies (beyond strictly necessary cookies) and/or analytics in order to be transparent to our users and to reduce our exposure to legal risks from changes in EU and US privacy law over the last several years.

Also see: https://mozilla-hub.atlassian.net/wiki/spaces/EN/pages/538050566/Cookie+Banner+Implementation+On+Mozilla.org

### PR DESCRIPTION
**Pull request by Iinh**
Fixes #2692.

Preview:
CleanShot 2024-02-28 at 13 59 03@2x

### CODE REVIEW

**Comment by edugfilho** [comment]
The layout looks good to me! I see it's in draft and the actual preferences are not coded in yet. What are next steps to get those?
Also, wasn't there a lib for it? I guess it's not a lot of coding, so maybe not needed.

File: src/components/CookieBanner.svelte [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      settingsOpen = !settingsOpen;
+    }
+  </script>
+  
+  {#if !consentGiven}
+  <div class="cookie-banner">
+    <p>We use cookies to enhance your experience. By continuing to visit this site you agree to our use of cookies.</p>
+    <button on:click={toggleSettings}>Manage cookie settings</button>
+    <button on:click={giveConsent}>Accept</button>
+    <button on:click={rejectCookies}>Reject</button>
```
Comment (edugfilho) [inline src/components/CookieBanner.svelte:84]: what is supposed to happen if all cookies get rejected?

**Comment by Iinh** [comment]
> The layout looks good to me! I see it's in draft and the actual preferences are not coded in yet. What are next steps to get those?

Thanks! I filled out the request form a while ago and am still waiting for guidance from Marketing/Legal, to decide what the final setting is.

File: src/components/CookieBanner.svelte [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    background-color: #e9ecef;
+    padding: 10px;
+    border-radius: 5px;
+  }
+</style>
+
+{#if !consentGiven}
+  <div class="cookie-banner">
+    <p>
+      We use cookies to enhance your experience, by continuing to visit this
```
Comment (edugfilho) [inline src/components/CookieBanner.svelte]:
```suggestion
      We use cookies to analyze our traffic, by continuing to visit this
```

**Review by edugfilho — APPROVED** [review]
r+wc
the one bit I would change is that the cookies in this case are not really to enhance the user experience but to analyze our traffic, so I made a suggestion there!

**Pull request merged**